src/main/scala/com/lunatech/pointingpoker/actors/RoomManager.scala

```diff
@@ -0,0 +1,119 @@
+  final case class CompleteWS()                                        extends Command
+  final case class ConnectToRoom(message: WSMessage, user: UntypedRef) extends Command
+
+  sealed trait Response
+  final case class RoomId(value: String) extends Response
+
+  val InitialVoteState  = false
+  val InitialEstimation = ""
+
+  final case class RoomManagerData(rooms: Map[UUID, ActorRef[Room.Command]])
+  object RoomManagerData {
+    val empty: RoomManagerData = RoomManagerData(rooms = Map.empty[UUID, ActorRef[Room.Command]])
+  }
+
+  def apply(): Behavior[Command] =
+    Behaviors.setup[Command] { _ =>
+      receiveBehaviour(RoomManagerData.empty)
+    }
+
+  private[actors] def receiveBehaviour(data: RoomManagerData): Behavior[Command] =
+    Behaviors
+      .receive[Command] { (context, message) =>
+        message match {
+          case CreateRoom(replyTo) =>
+            val roomId    = UUID.randomUUID()
+            val roomActor = createRoom(roomId, context)
+            val newData   = RoomManagerData(data.rooms + (roomId -> roomActor))
+
+            context.watch(roomActor)
+            replyTo ! RoomId(roomId.toString)
+            receiveBehaviour(newData)
+          case ConnectToRoom(message, user) =>
+            data.rooms
+              .get(message.roomId)
+              .fold {
+                val roomActor = createRoom(message.roomId, context)
+                context.watch(roomActor)
+                val newData = RoomManagerData(data.rooms + (message.roomId -> roomActor))
+                roomActor ! Room.Join(
+                  Room
+                    .User(message.userId, message.extra, InitialVoteState, InitialEstimation, user)
+                )
+                receiveBehaviour(newData)
+              } { room =>
+                room ! Room.Join(
+                  Room
+                    .User(message.userId, message.extra, InitialVoteState, InitialEstimation, user)
+                )
+                Behaviors.same
+              }
+          case IncomeWSMessage(message) =>
+            data.rooms.get(message.roomId).foreach(handleIncomeMessage(_, message, context))
+            Behaviors.same
+          case UnsupportedWSMessage =>
+            context.log.error("UnsupportedWSMessage received")
+            Behaviors.same
+          case WSCompleted(roomId, userId) =>
+            data.rooms.get(roomId).foreach(room => room ! Room.Leave(userId))
+            Behaviors.same
+          case WSFailure(t) =>
+            context.log.error("WSFailure: {}", t)
+            Behaviors.same
+          case CompleteWS() =>
+            context.log.error("CompleteWS: should never be received")
+            Behaviors.same
+        }
+      }
+      .receiveSignal {
+        case (_, Terminated(ref)) =>
+          val newData = RoomManagerData(data.rooms.filter {
+            case (_, roomRef) => roomRef != ref
+          })
+          receiveBehaviour(newData)
+      }
```

Comment:
Probably just style preference, but I'd extract out the filter here. I think this may read nicer
```suggestion
      .receiveSignal {
        case (_, Terminated(ref)) =>
          val leftoverRooms = data.rooms.filterNot { case (_, roomRef) => roomRef == ref }
          receiveBehaviour(RoomManagerData(leftoverRooms))
      }
```
wdyt?

Reply:
Done, also keeping the same approach as the other filterNot for consistency